internal/types/service_package.go

```diff
@@ -24,6 +31,7 @@ type ServicePackageEphemeralResource struct {
 	Factory  func(context.Context) (ephemeral.EphemeralResourceWithConfigure, error)
 	TypeName string
 	Name     string
+	Region   *ServicePackageResourceRegion
```

Comment:
In this and other specifications, `Region` is always configured by the generator. Consider using `ServicePackageResourceRegion` instead of `* ServicePackageResourceRegion`

Reply:
Relates https://github.com/hashicorp/terraform-provider-aws/pull/42186.